### PR TITLE
refactor: Update tests for batchUploader, beaconUpload, session bugfix, kit-blocking, persistence

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
Update tests for batchUploader, beaconUpload, and session bugfix

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - {explain how this has been tested, and what, if any, additional testing should be done}

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5342

### CODE REVIEW

File: test/src/tests-batchUploader.ts [hunk 1]
```diff
@@ -671,7 +672,7 @@ describe('batch uploader', () => {
             done();
         });
 
-        it('should purge events from Session Storage upon Batch Creation', (done) => {
+         it('should purge events from Session Storage upon Batch Creation', (done) => {
```
Comment (alexs-mparticle) [inline test/src/tests-batchUploader.ts:675]: For some reason, there's a stray space here.

```suggestion
        it('should purge events from Session Storage upon Batch Creation', (done) => {
```

File: test/src/tests-batchUploader.ts [hunk 2]
```diff
@@ -862,7 +863,7 @@ describe('batch uploader', () => {
 
             // First upload is successful
             window.fetchMock.post(urls.eventsV3, 200, {
-                overwriteRoutes: false,
+                overwriteRoutes: true,
```
Comment (alexs-mparticle) [inline test/src/tests-batchUploader.ts:866]: Why is this being changed to true?

File: test/src/tests-temp-session-bug-fix.js [hunk 3]
```diff
@@ -303,4 +303,4 @@ describe('session bug fix test', function() {
 
         done();
     });
-});
+});
```
Comment (alexs-mparticle) [inline test/src/tests-temp-session-bug-fix.js:306]: I think you should undo this change.

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: test/src/tests-batchUploader.ts (hunk 2 above)
Comment (rmi22186) [reply test/src/tests-batchUploader.ts:866]: Weird.  for some reason it was failing when on true so i switched it...but seems to be working again.  I'll revert

**Review by alexs-mparticle — APPROVED** [review]
